render-test/manifest_parser.cpp

```diff
@@ -0,0 +1,392 @@
+        if (!vendorPathValue.IsString()) {
+            mbgl::Log::Warning(
+                mbgl::Event::General, "Invalid vendorPath is provoided inside the manifest file: %s", filePath.c_str());
+            return {};
+        }
+        auto path = std::string(vendorPathValue.GetString());
+
+        manifest.vendorPath = (getValidPath(manifest.manifestPath, vendorPathValue.GetString()) / "").string();
+    }
+    mbgl::filesystem::path baseTestPath;
+    if (document.HasMember("base_test_path")) {
+        const auto& testPathValue = document["base_test_path"];
+        if (!testPathValue.IsString()) {
+            mbgl::Log::Warning(
+                mbgl::Event::General, "Invalid testPath is provoided inside the manifest file: %s", filePath.c_str());
+            return {};
+        }
+        baseTestPath = getValidPath(manifest.manifestPath, testPathValue.GetString());
+    }
+    mbgl::filesystem::path probeTestPath;
+    if (document.HasMember("probe_test_path")) {
+        const auto& testPathValue = document["probe_test_path"];
+        if (!testPathValue.IsString()) {
+            mbgl::Log::Warning(
+                mbgl::Event::General, "Invalid testPath is provoided inside the manifest file: %s", filePath.c_str());
+            return {};
+        }
+        probeTestPath = getValidPath(manifest.manifestPath, testPathValue.GetString());
+    }
+    std::vector<mbgl::filesystem::path> expectationPaths{};
+    if (document.HasMember("expectation_paths")) {
+        const auto& expectationPathValue = document["expectation_paths"];
+        if (!expectationPathValue.IsArray()) {
+            mbgl::Log::Warning(mbgl::Event::General,
+                               "Invalid expectationPath is provoided inside the manifest file: %s",
+                               filePath.c_str());
+            return {};
+        }
+        for (const auto& value : expectationPathValue.GetArray()) {
+            if (!value.IsString()) {
+                return {};
+            }
+            expectationPaths.emplace_back(getValidPath(manifest.manifestPath, value.GetString()));
+            if (expectationPaths.back().empty()) {
+                return {};
+            }
+        }
+    }
+    std::vector<mbgl::filesystem::path> ignorePaths{};
+    if (document.HasMember("ignore_paths")) {
+        const auto& ignorePathValue = document["ignore_paths"];
+        if (!ignorePathValue.IsArray()) {
+            mbgl::Log::Warning(
+                mbgl::Event::General, "Invalid ignorePath is provoided inside the manifest file: %s", filePath.c_str());
+            return {};
+        }
+        for (const auto& value : ignorePathValue.GetArray()) {
+            if (!value.IsString()) {
+                return {};
+            }
+            ignorePaths.emplace_back(getValidPath(manifest.manifestPath, value.GetString()));
+            if (ignorePaths.back().empty()) {
+                return {};
```

Comment:
ditto and `nullopt` too

Reply:
Here log is not necessary, there is a log inside `getValidPath` in case empty path is returned.